tests_e2e/orchestrator/lib/agent_test_suite_combinator.py

```diff
@@ -220,18 +228,18 @@ def create_environment_list(self) -> List[Dict[str, Any]]:
                                 env_name=env_name,
                                 marketplace_image=marketplace_image,
                                 vhd=vhd,
+                                shared_gallery=shared_gallery,
                                 location=location,
                                 vm_size=vm_size,
                                 test_suite_info=test_suite_info)
                         shared_environments[env_name] = env
 
-                    if test_suite_info.template != '':
```

Comment:
We were adding vm_tags["templates"] twice: here and below @ line 342. As part of the fix, I ensured there is always an env["vm_tags"] so it is safe to use [] instead of get() plus check on None.